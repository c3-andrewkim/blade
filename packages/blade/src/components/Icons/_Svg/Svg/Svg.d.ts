export type SvgProps = {
  fill?: string;
  height?: number;
  viewBox?: string;
  width?: number;
  xmlns?: string;
};

const Svg: React.ComponentType<SvgProps>;

export default Svg;
