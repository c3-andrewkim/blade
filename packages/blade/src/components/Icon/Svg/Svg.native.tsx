import React from 'react';
import type { ReactElement } from 'react';
import { Svg as SvgNative } from 'react-native-svg';
import type { SvgProps } from './Svg.d';

const Svg = (props: SvgProps): ReactElement => {
  return <SvgNative {...props} />;
};

export default Svg;
