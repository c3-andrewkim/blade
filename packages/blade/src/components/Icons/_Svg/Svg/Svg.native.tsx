import React from 'react';
import type { ReactElement } from 'react';
import { Svg as SvgNative } from 'react-native-svg';
import type { SvgProps } from './Svg.d';

const Svg = ({ children, height, viewBox, width, fill, xmlns }: SvgProps): ReactElement => {
  return <SvgNative {...{ children, height, viewBox, width, fill, xmlns }} />;
};

export default Svg;
