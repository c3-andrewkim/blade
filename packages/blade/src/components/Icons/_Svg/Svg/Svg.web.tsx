import React from 'react';
import type { ReactElement } from 'react';
import type { SvgProps } from './Svg.d';

const Svg = (props: SvgProps): ReactElement => {
  return <svg {...props} />;
};

export default Svg;
