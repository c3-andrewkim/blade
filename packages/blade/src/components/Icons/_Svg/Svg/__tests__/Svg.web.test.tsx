import React from 'react';
import Svg from '..';
import renderWithTheme from '../../../../../_helpers/testing/renderWithTheme.web';

describe('<Svg />', () => {
  it('should render html svg component', () => {
    const { container } = renderWithTheme(
      <Svg height="20px" width="20px" viewBox="0 0 24 24" fill="none" />,
    );
    expect(container).toMatchSnapshot();
  });
});
