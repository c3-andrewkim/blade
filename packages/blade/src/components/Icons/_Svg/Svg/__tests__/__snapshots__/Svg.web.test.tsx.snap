// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Svg /> should render html svg component 1`] = `
<div>
  <svg
    fill="none"
    height="20px"
    viewBox="0 0 24 24"
    width="20px"
  />
</div>
`;
